### PR TITLE
fix: overlay content

### PR DESCRIPTION
**Pull request by snitin315**
- [X] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

Yes

### Motivation / Use-Case
Fix https://github.com/webpack/webpack-dev-server/issues/3552

#### Before 

Screenshot 2021-07-22 at 4 44 46 PM

#### After

Screenshot 2021-07-22 at 4 51 23 PM

### Breaking Changes
No

### Additional Info
No

### CODE REVIEW

File: client-src/overlay.js [hunk 1]
```diff
@@ -138,7 +137,7 @@ function show(messages, type) {
 
       // Make it look similar to our terminal.
       const errorMessage = message.message || messages[0];
-      const text = ansiHTML(encode(errorMessage));
```
Comment (alexander-akait) [inline client-src/overlay.js:141]: It is not safe, here can be XSS

Comment (snitin315) [reply client-src/overlay.js:141]: Maybe we need [decode(text, {level: 'html5'})](https://github.com/mdevils/html-entities#decodetext-options) afterwards

Comment (alexander-akait) [reply client-src/overlay.js:141]: hm, maybe, just try it

**Review by alexander-akait — COMMENTED** [review]
Also we need test it

**Comment by snitin315** [comment]
> Also we need test it

WIP on tests

**Comment by snitin315** [comment]
Ready for review. /cc @alexander-akait

**Comment by alexander-akait** [comment]
Thanks, will look at near future

**Comment by alexander-akait** [comment]
I was wrong about XSS due `Creates a new Text node. This method can be used to escape HTML characters.` https://developer.mozilla.org/en-US/docs/Web/API/Document/createTextNode

**Comment by alexander-akait** [comment]
Fixed, anyway thanks

**Review by alexander-akait — APPROVED** [review]
Less deps :star:

**Comment by joacub** [comment]
the correct fix is this ====>
`var errorMessage = message.message || messages[0];
      var text = ansiHTML(errorMessage);
      var messageTextNode = document.createElement('div');`

html still showing bad, with this looks perfect, see the imagen bellow:

image

**Comment by alexander-akait** [comment]
It is xss, no it is not good

**Comment by alexander-akait** [comment]
Current solution has the same output

**Comment by joacub** [comment]
> Current solution has the same output

Impossible, html in a text node will be show up like entities….

**Review by joacub — COMMENTED** [review]
Html in text node will appear like `<span>raw</span>`

**Comment by alexander-akait** [comment]
Please provide screenshot, again, always use the issue template, our tests are find

**Comment by joacub** [comment]
> Please provide screenshot, again, always use the issue template, our tests are fine

Did you test with html content ?

**Comment by alexander-akait** [comment]
We should not accept HTML content from error/warning message, otherwise you can inject `<script>` and do XSS

**Comment by joacub** [comment]
Look is very simple, text node can no allocate html only text….

**Comment by alexander-akait** [comment]
Because it is XSS

**Comment by joacub** [comment]
> Because it is XSS

Inject xss to my self 🤣

**Comment by alexander-akait** [comment]
other plugin which you can install and use can emit HTML content and steel your environments, it is not joke

**Comment by joacub** [comment]
Perfect, is a 1% of the time that can happen maybe less, there is other solution to show correctly this? cause is impossible to read anything

**Comment by alexander-akait** [comment]
No one plugin should not emit HTML content in warning/error messages

**Comment by alexander-akait** [comment]
In your screenshot only text content and it should be good in overlay

**Comment by joacub** [comment]
> No one plugin should not emit HTML content in warning/error messages

Eslint do and do nice and read is bestifully, so best way is disable this not useful overlay

**Comment by alexander-akait** [comment]
> Eslint do and do nice and read is bestifully

HTML content? don't talk nonsense, eslint uses only ansi colors, I don't see HTML content

**Comment by joacub** [comment]
> > Eslint do and do nice and read is bestifully
> 
> HTML content? don't talk nonsense, eslint uses only ansi colors (yes, you can write custom output, but I don't see it is good idea), I don't see HTML content

Eslint return html you check in the other issue the screenshot with html …..

**Comment by alexander-akait** [comment]
No HTML in any place

**Comment by joacub** [comment]
Loook this

**Comment by alexander-akait** [comment]
Okay, please open a new issue with **reproducible test repo** and format this as issue template, original problem with html entries was fixed, what is why I ask ALWAYS use the issue template

**Pull request merged**